app/timer.hoon

```diff
@@ -0,0 +1,124 @@
+/+  *server
+/=  tile-js
+  /^  octs
+  /;  as-octs:mimes:html
+  /:  /===/app/timer/js/tile
+  /|  /js/
+      /~  ~
+  ==
+/=  timer-png
+  /^  (map knot @)
+  /:  /===/app/timer/img  /_  /png/
+=,  format
+::
+|%
+:: +move: output effect
+::
++$  move  [bone card]
+:: +card: output effect payload
+::
++$  poke
+  $%  [%noun [@tas path @t]]
+  ==
+::
++$  card
+  $%  [%poke wire dock poke]
+      [%http-response =http-event:http]
+      [%connect wire binding:http-server term]
+      [%diff %json json]
+      [%wait wire @da]
+      [%rest wire @da]
+  ==
+::
+--
+::
+|_  [bol=bowl:gall tim=@da]
+::
+++  this  .
+::
+++  bound
+  |=  [wir=wire success=? binding=binding:http-server]
+  ^-  (quip move _this)
+  [~ this]
+::
+++  prep
+  |=  old=(unit tim=@da)
+  ^-  (quip move _this)
+  =/  lismov/(list move)  %+  weld
+    `(list move)`[ost.bol %connect / [~ /'~timer'] %timer]~
+    `(list move)`[ost.bol %poke /timer [our.bol %launch] [%noun [%timer /tile '/~timer/js/tile.js']]]~
+  :-  lismov
+  ?~  old
+    this
+  %=  this
+    tim  tim.u.old
+  ==
+::
+++  peer-tile
+  |=  pax=path
+  ^-  (quip move _this)
+  ?:  =(tim *@da)
+    [[ost.bol %diff %json [%s '']]~ this]
+  [[ost.bol %diff %json [%s (scot %da tim)]]~ this]
+::
+++  send-tile-diff
+  |=  jon=json
+  ^-  (list move)
+  %+  turn  (prey:pubsub:userlib /tile bol)
+  |=  [=bone ^]
+  [bone %diff %json jon]
+::
+++  poke-json
+  |=  jon=json
+  ^-  (quip move _this)
+  ?.  ?=(%s -.jon)
+    [~ this]
+  =/  str/@t  +.jon
+  ?:  =(str 'start')
+    =/  data/@da  (add now.bol ~s10)
+    :-  %+  weld
+      `(list move)`(send-tile-diff [%s (scot %da data)])
+      `(list move)`[ost.bol %wait /timer data]~
```

Comment:
Backstep. Also, unless the order is important, just cons the `%wait` onto the head of the list produced by `+send-tile-diff`.

Reply:
fixed